root/components/RelationshipsTable.js

```diff
@@ -0,0 +1,158 @@
+/*
+ * @flow
+ * Copyright (C) 2019 MetaBrainz Foundation
+ *
+ * This file is part of MusicBrainz, the open internet music database,
+ * and is licensed under the GPL version 2, or (at your option) any
+ * later version: http://www.gnu.org/licenses/gpl-2.0.txt
+ */
+
+import * as React from 'react';
+
+import EntityLink from '../static/scripts/common/components/EntityLink';
+import loopParity from '../utility/loopParity';
+import generateRelationshipAppearancesList
+  from '../utility/generateRelationshipAppearancesList';
+import ArtistCreditLink
+  from '../static/scripts/common/components/ArtistCreditLink';
+import commaList from '../static/scripts/common/i18n/commaList';
+import formatDatePeriod
+  from '../static/scripts/common/utility/formatDatePeriod';
+import formatTrackLength
+  from '../static/scripts/common/utility/formatTrackLength';
+
+type Props = {|
+  +entity: CoreEntityT,
+  +heading: string,
+  +showCredits: boolean,
+|};
+
+const RelationshipsTable = ({
+  entity,
+  heading,
+  showCredits,
+}: Props) => {
+  const appearances = generateRelationshipAppearancesList(entity);
+  if (!appearances || appearances.length === 0) {
```

Comment:
Since appearances is an object, `appearances.length` will be undefined (unless we add a relationship type named "length" :)).

`Object.keys(appearances).length` should work. Maybe do `const relationshipTypes = Object.keys(appearances);` first since you also need the keys below where it loops over them.

(This should fix a bug where an empty table is displayed.)